src/v2/services/implementations/DappStakingService.ts

```diff
@@ -91,13 +104,69 @@ export class DappStakingService implements IDappStakingService {
   ): Promise<void> {
     Guard.ThrowIfUndefined('contractAddress', contractAddress);
     Guard.ThrowIfUndefined('stakerAddress', stakerAddress);
+    const isH160 = isValidEvmAddress(stakerAddress);
 
-    const stakeCall = await this.dappStakingRepository.getBondAndStakeCall(contractAddress, amount);
-    await this.wallet.signAndSend({
-      extrinsic: stakeCall,
-      senderAddress: stakerAddress,
-      successMessage,
-    });
+    if (isH160) {
```

Comment:
I suggest to create another dapp staking repository
`export class EvmDappStakingService implements IDappStakingService`
 which will handle EVM calls and inject it to the portal if user chooses Metamask (a wallet services are handled the same way). By doing this you will keep `DappStakingService` clean and simple.